### DISCUSSION TITLE
Access all github emails

### DISCUSSION THREAD

**Discussion by cfanoulis** [nextauthjs/next-auth#1109] [Help]
Hello! Is there any way to access all the email addresses from a github account without hitting the API, from the session object? thanks

**Comment by balazsorban44** [comment]
Hi there, please be more specific about what you want to achieve, so we can help you! I'm not sure I totally understand your question.

**Reply by cfanoulis** [reply]
Nevermind, I ended up using the github api

**Reply by cfanoulis** [reply]
if only I knew how to close this 😅

**Comment by iaincollins** [comment]
For the record, I think there is an odd situation with GitHub,in that if you have privacy settings enabled on your account then your email address isn't passed through on sign in (even though it is for most users with the default scope that is configured in the provider) but as I understand it in this case it can still be requested by an additional API call.

This is very similar to how LinkedIn works - IIRC it always requires you to call an API endpoint explicitly to get an email address for a user, as it never returns them in the default profile object.

I was wondering if perhaps we could facilitate calling the API to get this by passing additional options to the `profile` callback in the provider? I'm not sure off hand what it needs but I think something like the Access Token.

**Reply by balazsorban44** [reply]
Adding Instagram has a similar issue as well! #817
